roachprod: Remove outdated license warning message

spilchen opened the pull request:
Previously, starting a cluster with `roachprod` may display the following warning:
```
COCKROACH_DEV_LICENSE unset: enterprise features will be unavailable
```

With the deprecation of the core license, enterprise features are no longer disabled. Instead, clusters without a license are subject to throttling only after the 7-day grace period expires.

To address this, roachprod now automatically installs a license valid for one month. This prevents unnecessary warnings in the DB console and client notices about potential throttling.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135326)
<!-- Reviewable:end -->

rafiss reviewed (approved):
lgtm, though we could also alter the message so it mentions the grace period. (perhaps we can see if TestEng thinks if that would be helpful or not)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135326#-:-OC07VT2CY9AQ94IKW4R:bupzb6h)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @nameisbhaskar and @srosenberg)



<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
Initially, I considered updating the message, but the old message is quite misleading. If you set a license using `COCKROACH_DEV_LICENSE`, it's only applied the first time the cluster is started. Subsequent runs of `roachprod start` do not modify the license, but the message is still displayed. Mentioning the grace period in this context could add to the confusion. Instead, we should rely on client NOTICE messages and DB console banners to provide a more accurate indication of when throttling might begin. But I'll ask TestEng for their inputs.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135326#-:-OC3UX1FB6SsnKerjPPo:b-5s5x65)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @nameisbhaskar and @srosenberg)



<!-- Sent from Reviewable.io -->

spilchen commented:
I opted to change roachprod to install a license instead based on this slack discussion: https://cockroachlabs.slack.com/archives/C023S0V4YEB/p1732021509709429

srosenberg commented on pkg/roachprod/install/cockroach.go:
```diff
@@ -1294,9 +1295,13 @@ func (c *SyncedCluster) setClusterSettings(
+		license, _ = (&licenseccl.License{
+			Type:              licenseccl.License_Enterprise,
+			Environment:       licenseccl.Development,
+			ValidUntilUnixSec: timeutil.Now().AddDate(0, 1, 0).Unix(),
```
Is there any reason not to make this a perpetual license (e.g., `AddDate(1, 0, 0)`)? I'm thinking of some edge-cases, e.g., a long-running cluster.

spilchen replied:
We can, but since it’s an enterprise license, throttling does not apply. Once it expires, there are no product restrictions, aside from the DB console indicating that the license has expired.

srosenberg commented on pkg/roachprod/install/cockroach.go:
```diff
@@ -1294,9 +1295,13 @@ func (c *SyncedCluster) setClusterSettings(
+			Type:              licenseccl.License_Enterprise,
+			Environment:       licenseccl.Development,
+			ValidUntilUnixSec: timeutil.Now().AddDate(0, 1, 0).Unix(),
+		}).Encode()
```
We should still log (at INFO level) that a fresh license was generated, on the off chance that the user was expecting `config.CockroachDevLicense` to be non-empty.

srosenberg commented:
Kicked off a smoke test ([SELECT_PROBABILITY=0.4](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17843845?buildTab=log&logFilter=debug&focusLine=12&logView=linear) to make sure we didn't regress on anything.

spilchen committed:
roachprod: Remove outdated license warning message

Previously, starting a cluster with `roachprod` may display the following
warning:
COCKROACH_DEV_LICENSE unset: enterprise features will be unavailable

With the deprecation of the core license, enterprise features are no longer
disabled. Instead, clusters without a license are subject to throttling only
after the 7-day grace period expires.

To address this, roachprod now automatically installs a license valid
for one month. This prevents unnecessary warnings in the DB console and
client notices about potential throttling.

Epic: None
Release note: None

spilchen commented:
TFTR!

bors r+
